threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,84 @@
+| Disclosure of files          | Information disclosure | Mitigated     | High     | File System     | Files are accessed by attackers, this may lead to disclosure of notation config files and credential files | Use trusted infrastructure with right file permissions. Users can use credential store to securely store the credentials. Users can use notation plugin to access signing keys via remote KMS or a HSM device |
+| Malicious plugin             | Tampering              | Mitigated     | High     | Notation Plugin | A malicious plugin is installed, this may lead to arbitrary code being executed by attackers | Verify plugin package before installation or plugin is installed by authorized users |
+
+## Notation verify artifacts against signatures
+
+### Data flow
+
+A verifier uses Notation to verify artifacts against signatures stored in the file system or in a registry to ensure the authenticity and integrity of the artifacts before deploying the artifacts.
+
+The following diagram illustrates Notation verifying artifacts stored in a registry:
+
+![Notation verifies artifacts stored in registries](../media/notation-verify-remote.svg)
+
+The following diagram illustrates Notation verifying artifacts stored in a registry with a verification plugin:
+
+![Notation verifies artifacts stored in registries using verification plugin](../media/notation-verify-plugin.svg)
+
+The following diagram illustrates Notation verifying artifacts stored as OCI image layout in the file system:
+
+![Notation verifies artifacts as OCI image layout in the filesystem](../media/notation-verify-local.svg)
+
+The certificates trusted by the verifier are stored in Notation trust store in the file system. Users can download the certificates from KMS or request CA vendor for CA certificates.
+
+### Threats and Mitigation
+
+|   Title                      | STRIDE threat type     | Threat Status | Priority | Entity          | Description   | Mitigation   |
+| ---------------------------- | ---------------------- | ------------- | -------- | --------------- | ------------- | ------------ |
+| Inaccessible Registry        | Denial of Service      | Mitigated     | High     | Registry        | Registry is not able to service incoming requests or perform up to spec, thus users are unable to verify artifacts | Users verify signatures stored locally |
+| Serving tampered artifacts   | Tampering              | Mitigated     | High     | Registry        | Data stored in registries are tampered, this may lead to malicious artifacts being consumed | Users verify artifacts before consuming |
+| Disclosure of signature info | Information disclosure | Mitigated     | High     | Registry        | Attackers eavesdrop on registries, this may lead to the disclosure of sensitive signature info | Users should interact with trusted registries via TLS or Users can sign artifacts stored in local file system and store signature locally |
```

Comment:
We are talking about `Verify` in this section, but I notice that `sign` is still being mentioned in the `Mitigation` part. 

Besides, an open discussion on `Disclosure of signature info`. I'm not sure what does `this may lead to the disclosure of sensitive signature info` mean in the process of `verify`? I thought we can assume that during distribution of a signed artifact, any verifier (including attackers) should be able to access to the [artifact + signatures]. But even the attackers have access to the signed artifact and its signatures, they cannot reverse calculate the signer's private key. Therefore, IMO `Disclosure of signature info` is not a threat in `verify`.
For example, a signer signed their artifact and published it on Docker hub so that anyone interested can pull and verify the artifact. 
/cc: @shizhMSFT

Reply:
I agree with @Two-Hearts to remove this from verifying scenario. @shizhMSFT any comments?

Reply:
Thanks @shizhMSFT for explanation. This comment was resolved then.